File: exercises/practice/high-scores/high_scores.zig
pub const HighScores = struct {
    // This struct, as well as its fields and methods, needs to be implemented.

    pub fn init(scores: []const i32) HighScores {
        _ = scores;
        @compileError("please implement the init method");
    }

    pub fn latest(self: *const HighScores) ?i32 {
        _ = self;
        @compileError("please implement the latest method");
    }

    pub fn personalBest(self: *const HighScores) ?i32 {
        _ = self;
        @compileError("please implement the personalBest method");
    }

    pub fn personalTopThree(self: *const HighScores) []const i32 {
        _ = self;
        @compileError("please implement the personalTopThree method");
    }
};

File: generators/exercises/high_scores.py
HEADER = "const HighScores = high_scores.HighScores;"


def gen_case(case):
    scores = ", ".join(map(str, case["input"]["scores"]))
    prop = case["property"]
    e = case["expected"]
    s = f"    const scores = &[_]i32{{ {scores} }};\n"
    if prop == "latest":
        s += f"    try testing.expectEqual({e}, HighScores.init(scores).latest());\n"
    elif prop == "personalBest":
        s += f"    try testing.expectEqual({e}, HighScores.init(scores).personalBest());\n"
    elif prop == "personalTopThree":
        exp = ", ".join(map(str, e))
        s += f"    try testing.expectEqualSlices(i32, &[_]i32{{ {exp} }}, HighScores.init(scores).personalTopThree());\n"
    return s
